lib/logflare/sources.ex

```diff
@@ -40,31 +41,50 @@ defmodule Logflare.Sources do
 
   @spec create_source(map(), User.t()) :: {:ok, Source.t()} | {:error, Ecto.Changeset.t()}
   def create_source(source_params, user) do
+    postgres_backend_adapter_url =
+      Application.get_env(:logflare, :postgres_backend_adapter, []) |> Keyword.get(:url)
+
+    source_params =
+      source_params
+      |> Enum.map(fn {k, v} -> {to_string(k), v} end)
+      |> Map.new()
+      |> then(&Map.put(&1, "v2_pipeline", !!postgres_backend_adapter_url))
```

Comment:
would be cleaner to do these value setting at changeset level

Reply:
good call, will change